### DISCUSSION TITLE
Is it possible to enforce lockfileVersion?

### DISCUSSION THREAD

**Discussion by MojoJojo86** [renovatebot/renovate#23039] [Request Help] [answered]
### How are you running Renovate?

Placeholder value, please select the correct response from the dropdown

### If you're self-hosting Renovate, tell us what version of Renovate you run.

Latest 35.139.1

### If you're self-hosting Renovate, select which platform you are using.

GitHub Enterprise Server

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

We run npm 8 and while it can understand v3 package lock files, we have hit some issues and would like to know if we can enforce the package lock versioning to be set to v2 rather than v3.

### What you tried so far.

We have identified adding v3 and npm 9 support was a relatively recent feature request.
Checked the docs to see if there was a flag we could use in the config to support this.

**Comment by rarkins** [comment]
Is this only an issue when getting lock file maintenance PRs? And is renovate choosing npm 9 or npm 8?

**Reply by MojoJojo86** [reply]
We are running npm 8 in our containers where we are hosting renovate so npm 8 is been chosen.

I imagine that's the only interaction we are worried about but we are just trying to identify if it's going to be an issue when we implement renovate across the business.

If our projects are all currently v2 lockfiles and then renovate bumps them to v3, we are just concerned about any impact here.

**Reply by rarkins** [reply]
If npm 8 is chosen and it defaults to v2 lock file then I don't think you have a problem?

**Reply by MojoJojo86** [reply]
Ok so can you select npm 8 to be used in renovate or does it auto use 9?

**Reply by rarkins** [reply, marked as answer]
You can configure constraints.npm in Renovate config if you want control it

**Reply by MojoJojo86** [reply]
Thank you very much :)
